### ISSUE TITLE
Quantity is converted to ndarray after scipy.signal.convolve

### ISSUE THREAD

**Issue by Motherboard** [hgrecco/pint#1109]
The following code produces ndarray in should_have_newton_units even though I would have liked it to keep the units:

```python
import pint
import scipy.signal
ureg = pint.UnitRegistry()
signal = ureg.Quantity(np.arange(10), ureg.newton)
should_have_newton_units = scipy.signal.convolve(signal, scipy.signal.window.gaussian(3, 1))
```

**Comment by keewis** [comment]
unfortunately, we can't do anything about that right now: unlike `numpy`, `scipy` doesn't provide a mechanism to dispatch its functions (at least I'm not aware of one). That means that these functions assume that it's fine to cast to `numpy.array` (which removes the units).

**Comment by jthielen** [comment]
> unfortunately, we can't do anything about that right now: unlike `numpy`, `scipy` doesn't provide a mechanism to dispatch its functions (at least I'm not aware of one). That means that these functions assume that it's fine to cast to `numpy.array` (which removes the units).

Indeed, we are stuck being unable to support SciPy functions until further progress is made upstream (perhaps through [NEP 37](https://numpy.org/neps/nep-0037-array-module.html) or [NEP 42](https://numpy.org/neps/nep-0042-new-dtypes.html). In the meantime, your best option is probably to write a wrapper function for your particular use-case using `@ureg.wraps` (an example with `scipy.optimize.fsolve` is here: https://github.com/hgrecco/pint/issues/1092).
